### DISCUSSION TITLE
How to save dynamic variable from body to environment?

### DISCUSSION THREAD

**Discussion by volodymyr-romaniv-day** [thunderclient/thunder-client-support#1166] [Q&A]
I would like to save dynamic variable from body to environment after request is sent, how could I do this?

I wrote the next scripts, but it seems to me that custom filter doesn't work with dynamic variables:
function temp(input) {
    console.log("test", input);
    tc.setVar("test", input);
}

![image](https://github.com/rangav/thunder-client-support/assets/116160067/3a0c5267-d6da-4756-a410-f03ea93337d6)

I need it because every time I need to create new item and check that it is created in the new list. For this I need to auto-generate id and save it to environment variable

**Comment by rangav** [comment]
Hi @volodymyr-romaniv-day you cannot apply filter to dynamic variable.

`{{#name}}` - correct use of dynamic variable

You can only apply filter to env variable - `{{name, test}}`

**Reply by volodymyr-romaniv-day** [reply]
@rangav, so currently it is not possible to auto-save dynamic variables?

**Reply by rangav** [reply]
no not possible, you can use env variable and custom filter to achieve the same.

**Reply by volodymyr-romaniv-day** [reply]
Thank you again, I haven't thought about it. And it is even better for me

I wrote small function, maybe it will be useful to someone 

```
function temp(input,key_name) {
    var random_int = Math.floor(Math.random() * 100000);
    
    console.log(key_name, random_int);

    if (Boolean(key_name)){
        tc.setVar(key_name, random_int);};
    return random_int;
}
```

![image](https://github.com/rangav/thunder-client-support/assets/116160067/3bbe4d84-0f13-48bf-aae4-4a027f8ae518)
![image](https://github.com/rangav/thunder-client-support/assets/116160067/81eddc82-8311-4b21-b797-e3c4691e40dd)
![image](https://github.com/rangav/thunder-client-support/assets/116160067/7dea8e09-8071-4a19-bd20-6be679a26b0c)

**Reply by rangav** [reply]
Good, thanks for sharing.
